torch/_meta_registrations.py

```diff
@@ -2934,205 +2935,137 @@ def meta_addbmm(self, batch1, batch2, *, beta=1, alpha=1):
+                f"The argument {i+1} of must be List[Tensor], " f"but got {type(arg)}."
+            ),
+        )
+        if i == 0:
+            nelem = len(arg)
+        else:
+            torch._check(
+                nelem > 0 and nelem == len(arg),
+                lambda: (
+                    "self and argument {i+1} must be non-empty and match in length, "
+                    f"but got {nelem} and {len(arg)}."
+                ),
+            )
 
-@register_meta(
-    [
-        aten._foreach_add.Tensor,
-    ]
-)
-def meta__foreach_binop_tensor(self, other, alpha=1):
-    torch._check(
-        isinstance(self, List),
-        lambda: f"The first argument must be List[Tensor], but got {type(self)}.",
-    )
-    torch._check(
-        isinstance(other, torch.Tensor),
-        lambda: f"The second argument must be Tensor, but got {type(other)}.",
-    )
-    return [torch.empty_like(s) for s in self]
 
+def _meta_foreach_out_of_place(*args, _scalar_op=None, _nlists=1, **kwargs):
+    _check_meta_foreach(*args, _nlists=_nlists, **kwargs)
+    nelem = len(args[0])
 
-@register_meta(
-    [
-        aten._foreach_add_.Tensor,
-    ]
-)
-def meta__foreach_binop__tensor(self, other, alpha=1):
-    torch._check(
-        isinstance(self, List),
-        lambda: f"The first argument must be List[Tensor], but got {type(self)}.",
-    )
-    torch._check(
-        isinstance(other, torch.Tensor),
-        lambda: f"The second argument must be Tensor, but got {type(other)}.",
-    )
+    result = []
+    for elem in range(nelem):
+        each_args = [args[i][elem] for i in range(_nlists)]
+        result.append(_scalar_op(*each_args, *args[_nlists:], **kwargs))
 
+    return result
 
-@register_meta(
-    [
-        aten._foreach_add_.Scalar,
-        aten._foreach_mul_.Scalar,
-        aten._foreach_sub_.Scalar,
-        aten._foreach_div_.Scalar,
-        aten._foreach_maximum_.Scalar,
-    ]
-)
-def meta__foreach_binop__scalar(self, scalar=1):
-    torch._check(
-        isinstance(self, List),
-        lambda: f"The first argument of must be List[Tensor], but got {type(self)}.",
-    )
 
+def _meta_foreach_inplace(*args, _scalar_op=None, _nlists=1, **kwargs):
+    _check_meta_foreach(*args, _nlists=_nlists, **kwargs)
+    _meta_foreach_out_of_place(*args, _scalar_op=_scalar_op, _nlists=_nlists, **kwargs)
+    return
 
-@register_meta(
-    [
-        aten._foreach_add.Scalar,
-        aten._foreach_div.Scalar,
-        aten._foreach_mul.Scalar,
-        aten._foreach_sub.Scalar,
-    ]
-)
-def meta__foreach_binop_scalar(self, scalar=1):
-    torch._check(
-        isinstance(self, List),
-        lambda: f"The first argument of must be List[Tensor], but got {type(self)}.",
-    )
-    return [torch.empty_like(s) for s in self]
 
+def register_meta_foreach(ops):
+    for op_name, nlists in ops:
+        for overload in ["default", "Scalar", "Tensor", "List", "ScalarList"]:
+            op = getattr(getattr(aten, op_name), overload, None)
```

Comment:
`OpOverloadPacket` already has a method that returns the list of overloads if I'm not mistaken.